sys/net/application_layer/emcute/emcute.c

```diff
@@ -63,22 +62,14 @@ static volatile int result;
 
 static inline uint16_t get_u16(const uint8_t *buf)
 {
-#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
-    return (uint16_t)((buf[0] << 8) | buf[1]);
-#else
-    return (uint16_t)((buf[1] << 8) | buf[0]);
-#endif
+    network_uint16_t ns = { .u16 = *((uint16_t *)buf) };
```

Comment:
hmm, I knew there was something off... I was looking for something that is as efficient as my manual solution I had here before. Any tips?